org.thecrossingchurch.CustomJobs/Jobs/HubspotIntegrationMatching.cs

```diff
@@ -0,0 +1,718 @@
+            Debug.WriteLine( string.Format( "Comparing {0} Top-Level Domains.", tlds.Length ) );
+
+            // Get HS API Token
+            string attrKey = dataMap.GetString("AttributeKey");
+            string attrValue = GlobalAttributesCache.Get().GetValue(attrKey);
+            key = Encryption.DecryptString(attrValue);
+
+            // Don't see this being used for anything...
+            var current_id = 0;
+
+            // Setup Excel
+            ExcelPackage excel = new ExcelPackage();
+            ExcelWorksheet worksheet = SetupWorksheet(excel);
+            var row = 2;
+
+            // Setup Context & Services
+            RockContext _context = new RockContext();
+            PersonService personService = new PersonService(_context);
+            PersonSearchKeyService personSearchKeyService = new PersonSearchKeyService(_context);
+
+            // Get list of all contacts from HubSpot
+            contacts = new List<HSContactResult>();
+            request_count = 0;
+            string apiUrl = "https://api.hubapi.com/crm/v3/objects/contacts?limit=100&properties=email,firstname,lastname,phone,rock_person_id,rock_record_status,has_potential_rock_match,createdate,lastmodifieddate";
+            GetContacts(apiUrl);
+            Debug.WriteLine("Contacts returned: " + contacts.Count());
+            WriteToLog(string.Format("Total Contacts to Match: {0}", contacts.Count()));
+
+            // 5 cases, record statistics for each
+            // HS person without person id we find a match for
+            int idSet = 0;
+            // HS person with a person id that no longer exists in Rock
+            int idRemoved = 0;
+            // HS person with a person id that exists in Rock today
+            int idCorrect = 0;
+            // HS person has no match in rock
+            int idEmpty = 0;
+            // Inserted new HS person that was only in rock
+            int idAdded = 0;
+
+            
+
+            // Process each contact in HubSpot
+            for ( var i = 0; i < contacts.Count(); i++ )
```

Comment:
It looks like `i` is just used for getting the nth contact. Could we update this to a `foreach` loop for clarity?